manifests/plugin/default_hostgroup.pp

```diff
@@ -1,4 +1,20 @@
-# Installs foreman_default_hostgroup plugin
-class foreman::plugin::default_hostgroup {
-  foreman::plugin {'default_hostgroup': }
+# = Foreman Default Hostgroup
+#
+# This class installs the default_hostgroup plugin and optionally manages the configuration file
+#
+# === Parameters:
+#
+# $hostgroups:: An hash of hostgroup names and facts to add to the configuration
+#
+class foreman::plugin::default_hostgroup (
+  Hash $hostgroups = {},
+){
+  if $hostgroups {
+    foreman::plugin {'default_hostgroup':
+      config => template('foreman/default_hostgroup.yaml.erb')
+    }
+  }
+  else {
+    foreman::plugin {'default_hostgroup': }
+  }
```

Comment:
I'd prefer something like:
```suggestion
  if $hostgroups {
    $config = template('foreman/default_hostgroup.yaml.erb')
  } else {
    $config = undef
  }

  foreman::plugin {'default_hostgroup':
    config => $config,
  }
```

Also wonder if `$hostgroups` doesn't always evaluate to true and you need an explicit empty check, but there's a test so that should be covered.